Description: Prevent build script clobbering compiler flags
 This is so that they can be completely controlled externally (eg hardening)
Author: James Cowgill <james410@cowgill.org.uk>
Last-Update: 2014-03-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -219,10 +219,6 @@
     message("-- Executables will be written into ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
     message("-- Shared Objects will be written into ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
 
-    if(CMAKE_COMPILER_IS_GNUCXX)
-        set(CMAKE_CXX_FLAGS "-g -O0")        ## No optimization, debug info
-    endif(CMAKE_COMPILER_IS_GNUCXX)
-
     ## In debug, only add NDEBUG if the user says so
     if (PREVENT_WX_ASSERTS MATCHES 1)
     message("-- Adding -DNDEBUG to definitions")
@@ -248,11 +244,6 @@
     message("-- Executables will be written into ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
     message("-- Shared Objects will be written into ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
 
-    set(CMAKE_CXX_FLAGS "-O2")        ## Optimize
-    if(CMAKE_COMPILER_IS_GNUCXX)
-        set(CMAKE_EXE_LINKER_FLAGS "-s")  ## Strip binary
-    endif(CMAKE_COMPILER_IS_GNUCXX)
-
     ## In release, add NDEBUG unless explicitly told not to
     if (NOT PREVENT_WX_ASSERTS MATCHES 0)
     message("-- Adding -DNDEBUG to definitions")
--- a/codelite_make/CMakeLists.txt
+++ b/codelite_make/CMakeLists.txt
@@ -20,7 +20,7 @@
 # we need wxWidgets flags to be set only for the c++ files, so we do it like this
 # by setting the CMAKE_CXX_FLAGS
 execute_process(COMMAND ${CL_WX_CONFIG} --cxxflags OUTPUT_VARIABLE WX_CXX_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
-set (CMAKE_CXX_FLAGS ${WX_CXX_FLAGS})
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WX_CXX_FLAGS}")
 
 # prepare list of files
 FILE(GLOB SRCS "*.cpp")
--- a/codelite_terminal/CMakeLists.txt
+++ b/codelite_terminal/CMakeLists.txt
@@ -17,7 +17,7 @@
 # we need wxWidgets flags to be set only for the c++ files, so we do it like this
 # by setting the CMAKE_CXX_FLAGS
 execute_process(COMMAND ${CL_WX_CONFIG} --debug=no --cxxflags OUTPUT_VARIABLE WX_CXX_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
-set (CMAKE_CXX_FLAGS ${WX_CXX_FLAGS})
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WX_CXX_FLAGS}")
 
 add_definitions(-DNDEBUG)
 
--- a/sdk/codelite_indexer/CMakeLists.txt
+++ b/sdk/codelite_indexer/CMakeLists.txt
@@ -20,7 +20,7 @@
 # we need wxWidgets flags to be set only for the c++ files, so we do it like this
 # by setting the CMAKE_CXX_FLAGS
 execute_process(COMMAND ${CL_WX_CONFIG} --cxxflags OUTPUT_VARIABLE WX_CXX_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
-set (CMAKE_CXX_FLAGS ${WX_CXX_FLAGS})
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WX_CXX_FLAGS}")
 
 # prepare list of files
 FILE(GLOB SRCS "*.cpp" "libctags/*.c" "network/*.cpp")
